src/drivers/serial/tegra194-tcu.c

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2023, NIO
+ *
+ * SPDX-License-Identifier: GPL-2.0-only
+ */
+
+#include <config.h>
+#include <stdint.h>
+#include <util.h>
+#include <machine/io.h>
+#include <plat/machine/devices_gen.h>
+
+#define TCU_TX_REG              (0)
+#define TX_NUM_BYTES_FIELD_BIT  (24)
+#define TX_FLUSH_BIT            (26)
+#define TX_INTR_TRIGGER_BIT     (31)
+#define UART_REG(mmio, x)       ((volatile uint32_t *)(mmio + (x)))
+
+#ifdef CONFIG_PRINTING
+void uart_drv_putchar(unsigned char c)
+{
+    uint32_t reg_val;
+
+    /* We are writing one byte */
+    reg_val = (uint32_t)(1UL << TX_NUM_BYTES_FIELD_BIT);
```

Comment:
Any reason you're not using the BIT macro here?

Also, it would be nicer to move all the BIT() into the define itself instead of at the user's sites.

Reply:
Using `BIT()` seems semantically wrong here, because this is setting a shifted value.

Reply:
In that case, the name of `TX_NUM_BYTES_FIELD_BIT` is misleading, it should be called `TX_NUM_BYTES_FIELD_SHIFT`.

Reply:
Ah yes, I'll change that.